### PR TITLE
Reupload bug fixes

### PR DESCRIPTION
**Pull request by aaronweaver**
1. Fixed re-upload to validate a file is uploaded before processing findings.
2. In some cases reupload endpoints would return multiple endpoints, if this edge case occurs processing continues.

### CODE REVIEW

File: dojo/forms.py [hunk 1]
```diff
@@ -487,8 +487,8 @@ def clean(self):
         cleaned_data = super().clean()
         scan_type = cleaned_data.get("scan_type")
         file = cleaned_data.get("file")
-        if requires_file(scan_type) and not file:
-            raise forms.ValidationError('Uploading a Report File is required for {}'.format(scan_type))
+        if file is None:
```
Comment (valentijnscholten) [inline dojo/forms.py:490]: There seems to be some kind of exception for SONARQUBE API import scans which don't seem to require a file?

File: requirements.txt [hunk 2]
```diff
@@ -70,3 +70,4 @@ django-debug-toolbar-request-history==0.1.3
 vcrpy==4.1.1
 vcrpy-unittest==0.1.7
 git+https://github.com/radiac/django-tagulous.git@develop#egg=django-tagulous
+PyJWT==1.6.1
```
Comment (valentijnscholten) [inline requirements.txt:73]: bonus? :-)

**Comment by valentijnscholten** [comment]
resolved conflict

File: dojo/test/views.py [hunk 3]
```diff
@@ -622,6 +622,7 @@ def re_import_scan_results(request, tid):
                          "is highly recommended."
     test = get_object_or_404(Test, id=tid)
     scan_type = test.test_type.name
+    scan_type = test.test_type.name
```
Comment (valentijnscholten) [inline dojo/test/views.py:625]: I think one of these can be removed.

File: requirements.txt [hunk 4]
```diff
@@ -70,4 +70,4 @@ django-debug-toolbar-request-history==0.1.3
 vcrpy==4.1.1
 vcrpy-unittest==0.1.7
 git+https://github.com/radiac/django-tagulous.git@develop#egg=django-tagulous
-PyJWT==1.7.1
\ No newline at end of file
+PyJWT==1.7.1
```
Comment (damiencarol) [inline requirements.txt:73]: This modification should not be here. 🤔

Comment (madchap) [reply requirements.txt:73]: would be gone if rebased.

**Comment by madchap** [comment]
@aaronweaver can you please address the comments ASAP to see your bugfix eventually merged for tomorrow's release? Thanks.

**Review by madchap — APPROVED** [review]

**Review by devGregA — APPROVED** [review]

**Pull request merged**